src/run-with-limit.ts

```diff
@@ -0,0 +1,75 @@
+// Note: this is stolen from a broken Deno library (https://github.com/alextes/run_with_limit)
+// There is an open issue that should fix the issue and then we can just use the library itself.
+interface Runnable<A> {
+  fn: () => Promise<A>;
+  resolve: (value: A | PromiseLike<A>) => void;
+  reject: (reason?: any) => void;
+}
+
+/**
+ * Takes a number, setting the concurrency for the promise queue. Returns a set of functions to use the queue.
+ */
+export function makeRunWithLimit<A>(
```

Comment:
this function uses an unusual pattern (at least in our application) with the multiple functions return. Not sure why on earth dev do this stuff "well the language allows it to do this but the 3 major programming languages in this world would never be able to make this weird coding style, and nobody will be familiar with it, but I think its a good idea to do this"... That being said, personally, I would like to have a more common, and unique style in our codebase like e.g. wrapping this inside a class instead of this closure magic but if we anyways plan to replace this with the original lib I guess it does not make much sense so adapt this code towards another coding style.

Reply:
Yeah, it's a bit weird, but I guess it's okay. One thing you could do is have a look at https://github.com/sindresorhus/p-limit, which is the original vanilla NodeJS variant of this mechanism. It looks like they're just returning one function, which is maybe better? But I'm also not sure how you would solve this more beautifully. I'll go ahead and merge it.

Reply:
> Btw we also have the promises.ts file. These is code to allow to serialize the execution of promises.

Yes, I saw that. But that one will take a while. This makeRunWithLimit will allow you to run X promises in parallel, which is a lot nicer and faster. In your case with Azure I get it that you need this one-at-a-time execution, but for AWS it seems fine to do it like this.